genesis/engine/solvers/base_solver.py

```diff
@@ -36,6 +31,29 @@ def __init__(self, scene: "Scene", sim: "Simulator", options):
     def _add_force_field(self, force_field):
         self._ffs.append(force_field)
 
+    def build(self):
+        self._B = self._sim._B
+
+        g_np = np.asarray(
+            self._init_gravity if self._init_gravity is not None else self._sim._gravity,
+            dtype=np.float32,
+        )
+        g_np = np.repeat(g_np[None], self._B, axis=0)
+
+        self._gravity = ti.Vector.field(3, dtype=gs.ti_float, shape=self._B)
+        self._gravity.from_numpy(g_np)
+
+        if self._B == 1:
+            self._gravity.to_numpy = lambda _orig=self._gravity.to_numpy: _orig().squeeze(0)
```

Comment:
You cannot do that. Overwriting instance methods is very hacky and should only be used in last resort. Why would this be required in the first place? It is required by `set_gravity`, then just update `set_gravity` accordingly to avoid relying on this mechanism.